packages/client/src/clients/in-app/types.ts

```diff
@@ -0,0 +1,81 @@
+import { GenericData, PageInfo } from "@knocklabs/types";
+
+import { NetworkStatus } from "../../networkStatus";
+import { NotificationSource } from "../messages/interfaces";
+
+export interface CommonContentField {
+  key: string;
+  label: string;
+  rendered: string;
+  type: "text" | "markdown" | "boolean" | "textarea";
+  value: string;
+}
+
+export interface ButtonContentField {
+  key: string;
+  type: "button";
+  text: Exclude<CommonContentField, "key | type"> & {
+    key: "text";
+    type: "text";
+  };
+  action: Exclude<CommonContentField, "key | type"> & {
+    key: "action";
+    type: "text";
+  };
+}
+
+export type ContentField = CommonContentField | ButtonContentField;
+
+export interface InAppMessage<T = GenericData> {
+  __cursor: string;
+  id: string;
+  message_type: string;
+  schema_variant_key: string;
+  schema_version: string;
+  priority: number;
+  content: Record<string, ContentField>;
+  data: T | null;
+  inserted_at: string;
+  updated_at: string;
+  seen_at: string | null;
+  read_at: string | null;
+  interacted_at: string | null;
+  archived_at: string | null;
+  link_clicked_at: string | null;
+  source: NotificationSource;
+}
+
+export interface InAppMessagesQueryInfo {
+  items: InAppMessage[];
+  pageInfo: PageInfo;
+}
+
+export interface InAppStoreState {
+  networkStatus: NetworkStatus;
+  loading: boolean;
+  messages: Record<string, InAppMessage>;
+  queries: Record<string, InAppMessagesQueryInfo>;
+}
+
+export interface InAppMessageClientOptions {
+  order?: "asc" | "desc";
+  before?: string;
+  after?: string;
+  page_size?: number;
+  status?: "unread" | "read" | "unseen" | "seen" | "all";
```

Comment:
`engagement_status` as a list of statuses, and also double check on accept values.